pkg/reconciler/events/cloudevent/cloudeventsfakeclient.go

```diff
@@ -37,18 +40,15 @@ type FakeClientBehaviour struct {
 type FakeClient struct {
 	behaviour *FakeClientBehaviour
 	// Modelled after k8s.io/client-go fake recorder
-	Events chan string
-	// waitGroup is used to block until all events have been sent
-	waitGroup *sync.WaitGroup
+	events chan string
 }
 
 // newFakeClient is a FakeClient factory, it returns a client for the target
-func newFakeClient(behaviour *FakeClientBehaviour, expectedEventCount int) CEClient {
+func newFakeClient(behaviour *FakeClientBehaviour, expectedEventCount int) cloudevents.Client {
 	return FakeClient{
 		behaviour: behaviour,
 		// set buffersize to length of want events to make sure no extra events are sent
-		Events:    make(chan string, expectedEventCount),
-		waitGroup: &sync.WaitGroup{},
+		events: make(chan string, expectedEventCount),
```

Comment:
We set the client buffer to `expectedEventCount` which means that we cannot receive more events than expected, at least until we start reading messages from the channel.